.github/workflows/CI.yml

```diff
@@ -39,6 +39,25 @@ jobs:
         run: npx envinfo
       - name: Build
         run: ./configure && make -j8
+  build-from-tarball:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Environment Information
+        run: npx envinfo
+      - name: Make tarball
+        run: | 
+          export DISTTYPE=nightly
+          export DATESTRING=`date "+%Y-%m-%d"`
+          export COMMIT=xxxx
+          ./configure && make tar -j8
+      - name: Extract tarball
+        run: tar xzf *.tar.gz
+      - name: Build
+        run: |
+          export TAR_DIR=`basename *.tar.gz .tar.gz`
+          cd $TAR_DIR
+          ./configure && make -j8
```

Comment:
should we run the tests?

Reply:
I just followed what was done for the other GitHub action jobs, but I'm happy to add a test run if that is needed.

Reply:
I added test runs and it looks like there are some failures, but I'm not sure if those are known flakes.

Reply:
I think it will be much cleaner to remove the tests, since we have it in other place.

Reply:
Don't we still want to ensure that the build itself works?

Reply:
Yes, I missed this part. Test should be necessary.